frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelConversionWindowFilter.tsx

```diff
@@ -0,0 +1,74 @@
+import { InfoCircleOutlined } from '@ant-design/icons'
+import { pluralize } from 'lib/utils'
+import React from 'react'
+import { useActions, useValues } from 'kea'
+import { funnelLogic } from 'scenes/funnels/funnelLogic'
+import { FunnelConversionWindowTimeUnit } from '~/types'
+
+const TIME_INTERVAL_BOUNDS: Record<FunnelConversionWindowTimeUnit, number[]> = {
+    [FunnelConversionWindowTimeUnit.Minute]: [1, 1440],
+    [FunnelConversionWindowTimeUnit.Hour]: [1, 24],
+    [FunnelConversionWindowTimeUnit.Day]: [1, 365],
+    [FunnelConversionWindowTimeUnit.Week]: [1, 53],
+    [FunnelConversionWindowTimeUnit.Month]: [1, 12],
+}
+
+export function FunnelConversionWindowFilter(): JSX.Element {
+    const { conversionWindow } = useValues(funnelLogic)
+    const { setConversionWindow, loadResults } = useActions(funnelLogic)
+
+    const options = Object.keys(TIME_INTERVAL_BOUNDS).map((unit) => ({
+        label: pluralize(conversionWindow.funnel_window_interval, unit, `${unit}s`, false),
+        value: unit,
+    }))
+    const intervalBounds = TIME_INTERVAL_BOUNDS[conversionWindow.funnel_window_interval_unit]
+
+    return (
+        <div className="funnel-options-conversion-window">
+            <span className="funnel-options-conversion-window-label">
+                Conversion window limit{' '}
+                <Tooltip
+                    title={
+                        <>
+                            <b>Recommended!</b> Limit to users who converted within a specific time frame. Users who do
+                            not convert in this time frame will be considered as drop-offs.
+                        </>
+                    }
+                >
+                    <InfoCircleOutlined className="info-indicator" />
+                </Tooltip>
+            </span>
+            <Row className="funnel-options-conversion-window-inputs">
+                <InputNumber
+                    className="time-value-input"
+                    min={intervalBounds[0]}
+                    max={intervalBounds[1]}
+                    defaultValue={14}
+                    value={conversionWindow.funnel_window_interval}
+                    onChange={(funnel_window_interval) =>
+                        setConversionWindow({ funnel_window_interval: Number(funnel_window_interval) })
+                    }
+                    onBlur={loadResults}
+                    onPressEnter={loadResults}
+                />
+                <Select
+                    className="time-unit-input"
+                    defaultValue={FunnelConversionWindowTimeUnit.Day}
+                    dropdownMatchSelectWidth={false}
+                    value={conversionWindow.funnel_window_interval_unit}
+                    onChange={(funnel_window_interval_unit: FunnelConversionWindowTimeUnit) =>
+                        setConversionWindow({ funnel_window_interval_unit })
+                    }
+                    onBlur={loadResults}
```

Comment:
`onBlur` causes the graph to reload even if the value hasn't changed, which is kind of wasteful.

Reply:
Yeah we should try to only prompt for a recalculate or an automatic reload when the query state has changed definition.

Reply:
Copying over my response from (#5539)[https://github.com/PostHog/posthog/pull/5539#discussion_r686948106]

> Thanks for taking a stab at this! I think the motive behind moving it from onChange into onBlur was that we didn't want to load results every time the value was changed because that would be expensive. Perhaps we can move this back into onBlur and wrap it in a if(prevValue !== nextValue) {} (or handle this in the logic somewhere) to trigger only when the value has changed

Reply:
My newest changes will trigger a reload only when the interval value or interval unit are changed onblur, onenter but excluding onchange to avoid expensive recalculations.